dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/All.java

```diff
@@ -49,39 +56,73 @@ public void execute(final Namespace ns, final DockerfileGitHubUtil dockerfileGit
         }
         List<ProcessingErrors> imagesThatCouldNotBeProcessed = new LinkedList<>();
         AtomicInteger numberOfImagesToProcess = new AtomicInteger();
-        for (Map.Entry<String, JsonElement> imageToTag : imageToTagStore) {
-            numberOfImagesToProcess.getAndIncrement();
-            String image = imageToTag.getKey();
-            String tag = imageToTag.getValue().getAsString();
-            try {
-                PullRequests pullRequests = getPullRequests();
-                GitHubPullRequestSender pullRequestSender = getPullRequestSender(dockerfileGitHubUtil, ns);
-                GitForkBranch gitForkBranch = getGitForkBranch(image, tag, ns);
-
-                log.info("Finding Dockerfiles with the image name {}...", image);
-
-                Optional<List<PagedSearchIterable<GHContent>>> contentsWithImage =
-                        this.dockerfileGitHubUtil.findFilesWithImage(image, orgsToIncludeInSearch, gitApiSearchLimit);
-
-                if (contentsWithImage.isPresent()) {
-                    contentsWithImage.get().forEach(pagedSearchIterable -> {
-                        try {
-                            pullRequests.prepareToCreate(ns, pullRequestSender,
-                                    pagedSearchIterable, gitForkBranch, dockerfileGitHubUtil);
-                        } catch (IOException e) {
-                            log.error("Could not send pull request for image {}.", image);
-                            processErrors(image, tag, e, imagesThatCouldNotBeProcessed);
-                        }
-                    });
+        String store = ns.get(Constants.STORE);
+        Optional<Exception> failureMessage;
+        if (store.startsWith(s3Prefix)) {
```

Comment:
No need for an if statement. We should be calling our Store interface and have all this S3 stuff in the S3 implementation. Roughly speaking your code will look like this:

URI storeUri = new URI(store);
ImageTagStore imageTagStore;
switch (storeUri.getScheme()) {
  case("s3"):
        imageTagStore = new S3Store(...);
        break;
  default:
        imageTagStore = ... /* initialize like we did before */
}

imageTagStore.parseStoreToImagesMap(...)
  
}